### ISSUE TITLE
Support multi-path swaps on chain-abstraction sdk - Offchain

### ISSUE THREAD

**Issue by liu-zhipeng** [connext/monorepo#4237]
## Problem
https://github.com/connext/connext-integration/commit/5105962b149539d63cae6c1c9954f7efabcc630a

The Swapper contracts have been updated to support multipath swapping.
We need to change offchain code of chain-abstraction SDK as well. 

https://github.com/connext/monorepo/blob/688c8687cd9cedbf4746ab943fbc54597de23281/packages/agents/chain-abstraction/src/helpers/swapdata.ts#L59

## Impact
We can support all swapping which is from `any token` to `any token`.

## Proposed Solution
Basically we need to find the best path for given `from` and `to` asset address on SDK side.

https://github.com/joshstevens19/simple-uniswap-sdk

FIxes #4609 

## Acceptance Criteria
- [ ] Find the best swap path for Uniswap V2
- [ ] Find the best swap path for Uniswap V3

**Comment by rhlsthrm** [comment]
Working on this, investigating Uniswap SDKs.

**Comment by rhlsthrm** [comment]
Integrating Uniswap V3 SDK into chain abstraction SDK.

**Comment by rhlsthrm** [comment]
Still working on this, depri until subgraph issue taken care of.

**Comment by rhlsthrm** [comment]
Progressing, works with Uniswap, need to figure out Pancake.

**Comment by rhlsthrm** [comment]
Looking at Gnosis chain stuff. Need to figure out how to query HoneySwap properly, there is no SDK that we can find.

**Comment by rhlsthrm** [comment]
Honeyswap SDK is giving weird errors/responses.

**Comment by rhlsthrm** [comment]
Started writing tests, Honeyswap still an issue.

**Comment by rhlsthrm** [comment]
PR is created.

**Comment by prathmeshkhandelwal1** [comment]
https://github.com/connext/monorepo/pull/4609
